userspace/libscap/scap_procs.c

```diff
@@ -557,6 +557,11 @@ static int32_t scap_proc_add_from_proc(scap_t* handle, uint32_t tid, int parentt
 	}
 
 	//
+	// Gathers the exepath
+	//
+	snprintf(tinfo->exepath, target_res, "%s", filename);
```

Comment:
This invocation of `snprintf` looks suspicious. Why are you using `target_res` instead of the proper size of `exepath`? Also, isn't `filename` just the name of the symlink in `/proc`, while you instead need the file that is pointed by the link (`target_name`)?